applications/nrf_desktop/src/modules/Kconfig.hid_state

```diff
@@ -9,8 +9,14 @@ menu "HID State"
 config DESKTOP_HID_STATE_ENABLE
 	bool "Enable HID state"
 	depends on BT_PERIPHERAL
+	depends on DESKTOP_USE_DEFAULT_REPORT_DESCR
```

Comment:
This should not be added. HID reports implemented in our default report descriptor and handled in hid_state are standard report types for mouse and keyboard.
By adding this change you will take away user to provide more specific report descriptor (e.g. with vendor specific reports that are not handled by hid_state), or you will take away the possibility to handle standard keyboard and mouse reports by hid_state. Both options are wrong in my opinion.

The coupling between report descriptor and hid_state is known but this is not the way it should be resolved.

Reply:
Ok, I initially though that user could be able to remove the dependency and update source code of HID state together if needed (if user changed source code, he can also modify Kconfig to match new implementation). Still, I see this is misleading, so removed this part.